ops/_private/yaml.py

```diff
@@ -12,21 +12,32 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# autopep8 hates monkeypatching
+# autopep8: off
+# flake8: noqa
+
 """Internal YAML helpers."""
 
-import yaml
+import yaml as pyyaml
 
 
 # Use C speedups if available
-_safe_loader = getattr(yaml, 'CSafeLoader', yaml.SafeLoader)
-_safe_dumper = getattr(yaml, 'CSafeDumper', yaml.SafeDumper)
+_safe_loader = getattr(pyyaml, "CSafeLoader", pyyaml.SafeLoader)
+_safe_dumper = getattr(pyyaml, "CSafeDumper", pyyaml.SafeDumper)
 
 
 def safe_load(stream):
     """Same as yaml.safe_load, but use fast C loader if available."""
-    return yaml.load(stream, Loader=_safe_loader)
+    return pyyaml.load(stream, Loader=_safe_loader)
 
 
 def safe_dump(data, stream=None, **kwargs):
     """Same as yaml.safe_dump, but use fast C dumper if available."""
-    return yaml.dump(data, stream=stream, Dumper=_safe_dumper, **kwargs)
+    return pyyaml.dump(data, stream=stream, Dumper=_safe_dumper, **kwargs)
+
+
+pyyaml.safe_load = safe_load
+pyyaml.safe_dump = safe_dump
```

Comment:
So the whole reason that we have ops/_private/yaml.py is  because we *didn't* want to monkey patch upstream yaml, but we did want to have a consistent pattern within our code base.
Monkey patching other people's libraries is very much a here-be-dragons things to be doing, as it is a side effect, which means sometimes other people's code is going to work, and other times it won't, if things happen to be imported in a different order, or someone does:
`from yaml import safe_load`
rather than
```
import yaml
yaml.safe_load
```

Given everything that you've experienced so far, I'd actually be more inclined to say we should make Stored* *less* like a dict so people are clear about what it is  and isn't. You can't store arbitrary content in it (instances are not supported, etc)